policykit/integrations/slack/models.py

```diff
@@ -176,41 +174,107 @@ def handle_metagov_event(self, outer_event):
             new_api_action.community_origin = True
             new_api_action.is_bundled = False
             new_api_action.save()  # save triggers policy evaluation
-        else:
-            logger.debug(f"{event_type}: no PlatformAction created")
-
-        if event_type == "reaction_added":
-            event = outer_event["data"]
-            ts = event["item"]["ts"]
-
-            # check if this ts corresponds to a community_post action or bundle
-            action = PlatformAction.objects.filter(community=self, community_post=ts).first()
-            action_bundle = PlatformActionBundle.objects.filter(community=self, community_post=ts).first()
-            reaction_bool = reaction_to_boolean(event["reaction"])
-
-            if action is not None and reaction_bool is not None:
-                user, _ = SlackUser.objects.get_or_create(username=initiator, community=self)
-                logger.debug(f"Processing boolean Slack vote {reaction_bool} by {user}")
-                existing_vote = BooleanVote.objects.filter(proposal=action.proposal, user=user).first()
-                if existing_vote is not None:
-                    existing_vote.boolean_value = reaction_bool
-                    existing_vote.save()
-                else:
-                    BooleanVote.objects.create(proposal=action.proposal, user=user, boolean_value=reaction_bool)
-
-            elif action_bundle is not None and event["reaction"] in NUMBERS_TEXT.keys():
-                bundled_actions = list(action_bundle.bundled_actions.all())
-                num = NUMBERS_TEXT[event["reaction"]]
-                voted_action = bundled_actions[num]
-                user, _ = SlackUser.objects.get_or_create(username=initiator, community=self)
-                logger.debug(f"Processing numeric Slack vote {num} for action {voted_action} by user {user}")
-
-                existing_vote = NumberVote.objects.filter(proposal=voted_action.proposal, user=user).first()
-                if existing_vote is not None:
-                    existing_vote.number_value = num
-                    existing_vote.save()
-                else:
-                    NumberVote.objects.create(proposal=voted_action.proposal, user=user, number_value=num)
+            logger.debug(f"PlatformAction saved: {new_api_action.pk}")
+
+    def handle_metagov_process(self, process):
+        """
+        Handle a change to an ongoing Metagov slack.emoji-vote GovernanceProcess.
+        This function gets called any time a slack.emoji-vote associated with
+        this SlackCommunity gets updated (e.g. if a vote was cast).
+        """
+        assert process["name"] == "slack.emoji-vote"
```

Comment:
This is where the SlackCommunity tallies up votes and creates BooleanVote and NumberVote models for each. This gets called each time the governance process is updated. The process object looks like this:

```
{
  "id": 18,
  "name": "slack.vote",
  "community": "30ab507c-5f88-4194-95a0-4f583dd9a400",
  "status": "completed",
  "errors": {},
  "outcome": {
    "url": "https://testworkspace-i0j9034.slack.com/archives/C01G68P700Z/p1624566936002800",
    "channel": "C01G68P700Z",
    "message_ts": "1624566936.002800",
    "votes": {
      "yes": {
        "users": [
          "U025LEMKQP7",
           "U025LEMKQ88"
        ],
        "count": 2
      },
      "no": {
        "users": [
        ],
        "count": 0
      }
    }
  }
}
```